src/compat/db_json.cpp

```diff
@@ -2916,71 +3035,83 @@ db_json_sql_path_is_valid (std::string &sql_path)
       return false;
     }
   // start parsing path string by skipping dollar character
-  for (std::size_t i = 1; i < sql_path.length (); ++i)
+  std::size_t i = skip_whitespaces (sql_path, 1);
+  while (i < sql_path.length ())
     {
-      // to begin a next token we have only 2 possibilities:
+      // to begin a next token we have only 3 possibilities:
       // with dot we start an object name
       // with bracket we start an index
+      // with * we have the beginning of a '**' wildcard
       switch (sql_path[i])
 	{
 	case '[':
-	  end_bracket_offset = sql_path.find_first_of (']', ++i);
+	{
+	  std::size_t end_bracket_offset;
+	  i = skip_whitespaces (sql_path, i + 1);
+
+	  end_bracket_offset = sql_path.find_first_of (']', i);
 	  if (end_bracket_offset == std::string::npos)
 	    {
-	      // unacceptable
-	      assert (false);
 	      return false;
 	    }
-	  if (!db_json_path_is_token_valid_array_index (sql_path, i, end_bracket_offset))
+	  if (!db_json_path_is_token_valid_array_index (sql_path, allow_wildcards, i, end_bracket_offset))
 	    {
-	      // expecting a valid index
 	      return false;
 	    }
-	  // move to ']'. i will be incremented.
-	  i = end_bracket_offset;
+	  i = skip_whitespaces (sql_path, end_bracket_offset + 1);
 	  break;
-
+	}
 	case '.':
-	  i++;
-
-	  if (sql_path[i] == '"')
+	  i = skip_whitespaces (sql_path, i + 1);
+	  if (i == sql_path.length ())
 	    {
-	      i++;
-
-	      // right now this method accepts escaped quotes with backslash
-	      while (i < sql_path.length () && (sql_path[i] != '"' || sql_path[i - 1] == '\\'))
+	      return false;
+	    }
+	  switch (sql_path[i])
+	    {
+	    case '"':
+	      if (!db_json_path_is_token_valid_quoted_object_key (sql_path, i))
 		{
-		  i++;
+		  return false;
 		}
-
-	      if (i == sql_path.length ())
+	      break;
+	    case '*':
+	      if (!allow_wildcards)
 		{
 		  return false;
 		}
-	    }
-	  else
-	    {
-	      // we can have an object name without quotes only if the first character is a letter
-	      // otherwise we need to put in between quotes
-	      if (!std::isalpha (sql_path[i++]))
+	      i = skip_whitespaces (sql_path, i + 1);
+	      break;
+	    default:
+	      // unquoted object_keys
+	      if (!db_json_path_is_token_valid_unquoted_object_key (sql_path, i))
 		{
 		  return false;
 		}
+	      break;
+	    }
+	  break;
 
-	      while (i < sql_path.length () && (sql_path[i] != '.' && sql_path[i] != '['))
-		{
-		  i++;
-		}
+	case '*':
+	  // only ** wildcard is allowed in this case
+	  if (!allow_wildcards || (++i < sql_path.length () && sql_path[i] != '*'))
```

Comment:
I think the correct condition is:
```suggestion
	  if (!allow_wildcards || ++i >= sql_path.length () || sql_path[i] != '*')
```